build-ps/percona-server-5.7_builder.sh

```diff
@@ -0,0 +1,825 @@
+    sed -i "s:@@PERCONA_VERSION@@:${MYSQL_VERSION_EXTRA#-}:g" build-ps/percona-server.spec
+    sed -i "s:@@REVISION@@:${REVISION}:g" build-ps/percona-server.spec
+    sed -i "s:@@RPM_RELEASE@@:${RPM_RELEASE}:g" build-ps/percona-server.spec
+    sed -i "s:@@BOOST_PACKAGE_NAME@@:${BOOST_PACKAGE_NAME}:g" build-ps/percona-server.spec
+    cd ${WORKDIR}/percona-server
+    tar --owner=0 --group=0 --exclude=.bzr --exclude=.git -czf ${PSDIR}.tar.gz ${PSDIR}
+    
+    mkdir $WORKDIR/source_tarball
+    mkdir $CURDIR/source_tarball
+    cp ${PSDIR}.tar.gz $WORKDIR/source_tarball
+    cp ${PSDIR}.tar.gz $CURDIR/source_tarball
+    cd $CURDIR
+    rm -rf percona-server  
+    return
+}
+
+get_system(){
+    if [ -f /etc/redhat-release ]; then
+        RHEL=$(rpm --eval %rhel)
+        ARCH=$(echo $(uname -m) | sed -e 's:i686:i386:g')
+        OS_NAME="el$RHEL"
+        OS="rpm"
+    else
+        ARCH=$(uname -m)
+        OS_NAME="$(lsb_release -sc)"
+        OS="deb"
+    fi
+    return
+}
+
+install_deps() {
+    if [ $INSTALL = 0 ]
+    then
+        echo "Dependencies will not be installed"
+        return;
+    fi
+    if [ ! $( id -u ) -eq 0 ]
+    then
+        echo "It is not possible to instal dependencies. Please run as root"
+        exit 1
+    fi
+    CURPLACE=$(pwd)
+    
+    if [ "x$OS" = "xrpm" ]; then
+        RHEL=$(rpm --eval %rhel)
+        ARCH=$(echo $(uname -m) | sed -e 's:i686:i386:g')
+        add_percona_yum_repo
+        yum -y install http://www.percona.com/downloads/percona-release/redhat/0.1-4/percona-release-0.1-4.noarch.rpm || true
+        yum -y install epel-release
+        yum -y install git numactl-devel rpm-build gcc-c++ gperf ncurses-devel perl readline-devel openssl-devel jemalloc 
+        yum -y install time zlib-devel libaio-devel bison cmake pam-devel libeatmydata jemalloc-devel
+        yum -y install perl-Time-HiRes libcurl-devel openldap-devel unzip wget libcurl-devel 
+        yum -y install perl-Env perl-Data-Dumper perl-JSON MySQL-python perl-Digest perl-Digest-MD5 perl-Digest-Perl-MD5 || true
+        if [ ${RHEL} -lt 7 -a $(uname -m) = x86_64 ]; then
+            yum -y install percona-devtoolset-gcc percona-devtoolset-gcc-c++ percona-devtoolset-binutils 
+        fi
+        if [ "x$RHEL" = "x6" ]; then
+            yum -y install Percona-Server-shared-56  
+        fi
+    else
+        apt-get -y install dirmngr || true
+        add_percona_apt_repo
+      	apt-get update
+        apt-get -y install dirmngr || true
+        apt-get -y install lsb-release wget
+        export DEBIAN_FRONTEND="noninteractive"
+        export DIST="$(lsb_release -sc)"
+
+	    until sudo apt-get update; do
+    	    sleep 1
+            echo "waiting"
+        done
+        apt-get -y purge eatmydata || true
+        echo "deb http://jenkins.percona.com/apt-repo/ ${DIST} main" > percona-dev.list
```

Comment:
one more time? you have add_percona_apt_repo()

Reply:
done